design/volume-snapshot-data-movement/volume-snapshot-data-movement.md

```diff
@@ -0,0 +1,772 @@
+	// If DataMover is "" or "velero", the built-in data mover will be used.
+	// +optional
+	DataMover string `json:"datamover,omitempty"`
+```	
+SnapshotMoveData will be used to decide the Working Mode.  
+DataMover will be used to decide the data mover to handle the DUCR. DUCR's DataMover value is derived from this value.    
+
+As mentioned in the Plugin Data Movers section, the data movement information for a restore should be the same with the backup. Therefore, the working mode for restore should be decided by checking the corresponding Backup CR; when creating a DDCR, the DataMover value should be retrieved from the corresponding Backup Result.  
+
+## Logging
+The logs during the data movement are categorized as below:
+- Logs generated by Velero
+- Logs generated by DMPs
+- Logs generated by DMs
+
+For 1 and 2, the existing plugin mechanism guarantees that the logs could be saved into the Velero server log as well as backup/restore persistent log.  
+For 3, Velero leverage on DMs to decide how to save the log, but they will not go to Velero server log or backup/restore persistent log. For VBDM, the logs are saved in the node-agent server log.  
+
+## Installation
+DMs need to be configured during installation so that they can be installed. Plugin DMs may have their own configuration, for VGDM, the only requirement is to install Velero node-agent.  
+Moreover, the DMP is also required during the installation.  
+For example, to move CSI snapshot through VBDM, below is the installation script:  
+```
+velero install \
+    --provider \
+    --image \
+    --plugins velero/velero-plugin-for-csi:xxx \
+    --features=EnableCSI \
+    --use-node-agent \ 
+```
+
+## Upgrade
+For VBDM, no new installation option is introduced, so upgrade is not affected.
+If plugin DMs require new options and so the upgrade is affected, they should explain them in their own documents.
+
+## CLI
+As explained in the Working Mode section, we add one more flag ```snapshot-move-data``` to indicate whether the snapshot data should be moved.  
+As explained in the Plugin Data Movers section, we add one more flag ```data-mover``` for users to configure the data mover to move the snapshot data.  
+Example of backup command are as below.  
+
+Below CLI means to create a backup with volume snapshot data movement enabled and with VBDM as the data mover:  
+```
+velero backup create xxx --include-namespaces --snapshot-move-data
+```
+
+Below CLI has the same meaning as the first one:  
+```
+velero backup create xxx --include-namespaces --snapshot-move-data --data-mover velero
+```
+
+Below CLI means to create a backup with volume snapshot data movement enabled and with "xxx-plugin-dm" as the data mover:  
+```
+velero backup create xxx --include-namespaces --snapshot-move-data --data-mover xxx-plugin-dm
+```
+
+Restore command is kept as is.  
+
+
+
+
+[1]: ../unified-repo-and-kopia-integration/unified-repo-and-kopia-integration.md
+[2]: ../general-progress-monitoring.md
```

Comment:
Broken link; now at https://github.com/vmware-tanzu/velero/blob/main/design/Implemented/general-progress-monitoring.md (missing `implemented/` infix)

Reply:
Thanks for noticing this.
I tried to add the `Implemented` infix, but finally realized that we cannot do this. At the release time of 1.12, we will also move this design to the `Implemented` folder (the same to the unified-repo-and-kopia-integration folder), then all the links will be fixed.

Therefore, let's leave it as is for now, and it will be fixed itself at the release time.